src/internal/github/repository.go

```diff
@@ -0,0 +1,24 @@
+package github
+
+import (
+	"fmt"
+	"log/slog"
+)
+
+type Repository struct {
```

Comment:
Since this seems to be called sometimes without `log`, maybe adding explicit constructor(s), even if not exported, would be beneficial? That would document what the required parameters are.